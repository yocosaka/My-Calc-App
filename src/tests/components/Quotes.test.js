import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Quotes from '../../components/Quotes';

describe('Quotes', () => {
  test('renders Quotes component ', () => {
    render(<Quotes />);
    const text = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa voluptate quia perferendis, et esse dolor officia sit, blanditiis dignissimos, dolore provident quaerat ullam. Debitis, mollitia! Perspiciatis mollitia odit natus explicabo.';
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getByText(text)).toBeInTheDocument;
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Quotes />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
