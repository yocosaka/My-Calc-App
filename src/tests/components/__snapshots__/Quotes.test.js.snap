// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes should match with snapshot 1`] = `
<div
  className="main-content white"
>
  <h2
    className="main-title"
  >
    Quotes
  </h2>
  <p
    className="main-text"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Culpa voluptate quia perferendis, et esse dolor officia sit, blanditiis dignissimos, dolore provident quaerat ullam. Debitis, mollitia! Perspiciatis mollitia odit natus explicabo.
  </p>
</div>
`;
